backend/app/core/vault.py

```
@@ -0,0 +1,245 @@
+
+    @retry(
+        stop=stop_after_attempt(3),
+        wait=wait_exponential(multiplier=1, min=4, max=10),
+    )
+    async def health_check(self) -> bool:
+        """Check Vault health status."""
+        try:
+            async with self.client() as client:
+                response = await client.get(f"{self.vault_url}/v1/sys/health")
+                return response.status_code in [200, 429, 472, 473]  # All valid states
+        except Exception as e:
```

Ruff (0.13.1)

Do not catch blind exception: `Exception`

(BLE001)